core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -338,8 +343,10 @@ trait WorkspaceComponent {
     ): ReadAction[Seq[Workspace]] =
       loadWorkspaces(findByIdsQuery(workspaceIds), attributeSpecs)
 
-    def listByNamespaces(namespaceNames: Seq[RawlsBillingProjectName]): ReadAction[Seq[Workspace]] =
-      loadWorkspaces(findByNamespacesQuery(namespaceNames))
+    def listV2WorkspacesByIds(workspaceIds: Seq[UUID],
+                              attributeSpecs: Option[WorkspaceAttributeSpecs] = None
+    ): ReadAction[Seq[Workspace]] =
+      loadWorkspaces(findV2WorkspacesByIdsQuery(workspaceIds), attributeSpecs)
 
     def countByNamespace(namespaceName: RawlsBillingProjectName): ReadAction[Int] =
```

Comment:
I think it would be good to do an explicit query for the v2 workspaces, and if none exist, do a query for v1 workspaces.

1. If v2 workspaces exist, show existing message, "Project cannot be deleted because it contains workspaces".
2. If _only_ v1 workspaces exist, show "Project cannot be deleted because it contains v1 workspaces."

Reply:
I like the sound of that or possibly doing a group by to get the counts for v2 and v1 in a single transaction. Your solution would be easier to rip out later. The only other thing would be to modify the error message for v1 workspaces to indicate that users should contact support for help deleting the billing project

Reply:
Implemented in https://github.com/broadinstitute/rawls/pull/2214/commits/7e7b84e6047d8f98d304462af21994dd92571c10

Reply:
I looked into adding test coverage for the v1 case, but it was getting ugly. I tested it manually.